link/platform.go

```diff
@@ -0,0 +1,68 @@
+package link
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"runtime"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+const (
+	// Since 4.17, syscalls symbols are generated with the `__x64_` prefix.
+	// https://github.com/torvalds/linux/commit/d5a00528b58cdb2c71206e18bd021e34c4eab878
+	LinuxVersionWithPrefix = 4<<16 + 17<<8 + 0
+)
+
+var parseArchSyscall = regexp.MustCompile(`__(\w*)_sys_(\w*)`)
+var parseSyscall = regexp.MustCompile(`sys_(\w*)`)
+
+func ResolveSyscall(syscall string) (string, error) {
```

Comment:
Why do you want to support this API? The user should give us the "platform independent" syscall name, then we don't need all of this code.